website/blog/2019-11-04-binding-react-navigation-5-to-ui-kitten.md

```diff
@@ -0,0 +1,420 @@
+
+Usually, the drawer menu is available in app on the home screen, so let's add it to `HomeNavigator`. Open [./src/navigation/home.navigator.tsx](https://github.com/artyorsh/react-navigation-ex-demo/blob/complete-exmaples/src/navigation/home.navigator.tsx) file and paste the following code:
+
+```js
+import React from 'react';
+import { createDrawerNavigator } from '@react-navigation/drawer';
+import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
+import { TodoNavigator } from './todo.navigator';
+import { ProfileNavigator } from './profile.navigator';
+import { AppRoute } from './app-routes';
+import { BottomHomeScreen, DrawerHomeScreen, AboutScreen } from '../scenes/home';
+
+const Drawer = createDrawerNavigator();
+const BottomTab = createBottomTabNavigator();
+
+const HomeBottomNavigator = (): React.ReactElement => (
+  <BottomTab.Navigator tabBar={BottomHomeScreen}>
+    <BottomTab.Screen name={AppRoute.TODO} component={TodoNavigator}/>
+    <BottomTab.Screen name={AppRoute.PROFILE} component={ProfileNavigator}/>
+  </BottomTab.Navigator>
+);
+
+export const HomeNavigator = (): React.ReactElement => (
+  <Drawer.Navigator drawerContent={DrawerHomeScreen}>
+    <Drawer.Screen name={AppRoute.HOME} component={HomeBottomNavigator}/>
+    <Drawer.Screen name={AppRoute.ABOUT} component={AboutScreen}/>
+  </Drawer.Navigator>
+);
+```
+
+In this example, we've implemented a Drawer Navigator with `createDrawerNavigator` and used it to display on the Home screen. We have also added `AboutScreen` to demonstrate navigation directly from the Drawer menu.
+
+Just like Top/Bottom tab navigators, the drawer navigator also has a special property for declaring custom drawer view. Use a `drawerContent` property to pass the custom view to the navigator. Open [./src/scenes/home/drawer-home.component.tsx](https://github.com/artyorsh/react-navigation-ex-demo/blob/complete-exmaples/src/scenes/home/drawer-home.component.tsx) file and add the following code:
+
+```js
+import React from 'react';
+import { Drawer, DrawerElement, MenuItemType } from 'react-native-ui-kitten';
+import { AppRoute } from '../../navigation/app-routes';
+import { SafeAreaLayout, SaveAreaInset} from '../../components/safe-area-layout.component';
+import { InfoIcon, LogoutIcon } from '../../assets/icons';
+
+const drawerData: MenuItemType[] = [
+  { icon: InfoIcon, title: 'About' },
+  { icon: LogoutIcon, title: 'Log Out' },
+];
+
+export const DrawerHomeScreen = (props): DrawerElement => {
+
+  const onMenuItemSelect = (index: number): void => {
+    const { [index]: selectedItem } = drawerData;
+
+    switch (selectedItem.title) {
+      case 'Log Out':
+        props.navigation.navigate(AppRoute.AUTH);
```

Comment:
For stuff like logging out, instead of navigate, we recommend to do the action that actually logs the user out, such as removing the auth token from the store. If you're conditionally rendering screens for auth like in the docs, then everything is automatically handled.

Note that even if you use the navigate approach, you still need to clear the token even in a basic app. So it's extra work.